### PR TITLE
feat(v2): onBrokenMarkdownLinks config

### PR DESCRIPTION
**Pull request by AmyrAhmady**
## Motivation

Fixing issue #3652 

#### Little additional info: 

We have a website which uses Docusaurus v2, and recently, we are letting people to contribute to the project and translate existing pages to their own language in separate folders; This means relative markdown links are still there in newly translated pages while those pages might not exist in the language's folder (as I mentioned, those links are relatively written there)

An example of this behaviour would be this: [Click Here](https://wiki.open.mp/docs/translations/tr/scripting/functions/AddMenuItem#i%CC%87lgili-fonksiyonlar), and first item of that list does not exist. This was just an example that leads to printing only one error to stdout, for the real issue please refer to #3652

And at the end, there was a comment there saying this: https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-content-docs/src/index.ts#L314

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

I'd be lying if I say I've read it fully and carefully, but I tried to cherry pick needed parts but scrolling through, sorry :)

## Test Plan

Not sure how exactly we are supposed to test that, but I guess running an instance of Docusaurus with `"onBrokenLinks": "ignore"` in site configurations and a dummy doc page with a broken relative link would be fine.

Here's a screenshot (I've heard rumors it has bonus points!)
Docusaurus SS

## Related PRs

None I guess?

### CODE REVIEW

**Review by slorber — CHANGES_REQUESTED** [review]
Hi and thanks for completing my todo :) 
That looks pretty good overall.

There's for me one major problem: if you have 2 MD broken links and the throw config (recommended, default config), you might fail on first md broken link found. 
In such case only one broken link would be reported: the first one.

We'd rather find a way to aggregate all the md broken links here instead of failing fast asap? 
Maybe storing a mutable broken links list in the plugin, and reporting all the broken links once all the docs have been read?

If that's not possible, I'd rather have a 2nd option like siteConfig.onMarkdownBrokenLink or something, that would rather default to a warning, so that user might see all the md broken links in a single build attempt.

The important thing to me is that the user does not have to fail a build, fix 1 link, fail another build, fix another link etc... he should see all the broken links by default.

Does it make sense?

File: packages/docusaurus-plugin-content-docs/src/index.ts [hunk 1]
```diff
@@ -311,11 +315,13 @@ export default function pluginContentDocs(
         sourceToPermalink,
         versionsMetadata,
         onBrokenMarkdownLink: (brokenMarkdownLink) => {
-          // TODO make this warning configurable?
-          console.warn(
-            chalk.yellow(
-              `Docs markdown link couldn't be resolved: (${brokenMarkdownLink.link}) in ${brokenMarkdownLink.filePath} for version ${brokenMarkdownLink.version.versionName}`,
-            ),
+          if (siteConfig.onBrokenLinks === 'ignore') {
```
[the remark this answers is not in the document]
Comment (AmyrAhmady) [reply packages/docusaurus-plugin-content-docs/src/index.ts:318]: well I was actually doing it like this here: https://github.com/facebook/docusaurus/blob/master/packages/docusaurus/src/server/brokenLinks.ts#L206-L208

which is kind of logical, why go through execution of `reportMessage` scope when there's nothing to be executed at all?

**Comment by AmyrAhmady** [comment]
I have just pushed two commits, containing changes and features you've asked above :) 
What I did making broken markdown links independently configurable so we don't need to mix them up with actual broken links;
That being said, we now have a totally new option in site configs, called `onBrokenMarkdownLinks` and works exactly like `onBrokenLinks`, and its default value is set to `warn` 

Let me know if there's anything needed

File: packages/docusaurus-plugin-content-docs/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (AmyrAhmady) [reply packages/docusaurus-plugin-content-docs/src/index.ts]: Marking it as resolved as it is outdated, please read my comment above and let me know if I'm following the right path :)

File: website/versioned_docs/version-2.0.0-alpha.66/api/docusaurus.config.js.md [hunk 2]
```diff
@@ -108,6 +108,14 @@ The broken links detection is only available for a production build (`docusaurus
...
 :::
 
+### `onBrokenMarkdownLinks`
+
+- Type: `'ignore' | 'log' | 'warn' | 'error' | 'throw'`
+
+The behavior of Docusaurus, when it detects any broken markdown link.
+
+By default, it prints a warning, to let you know about your broken markdown link, but you can change this security if needed.
+
```
Comment (slorber) [inline website/versioned_docs/version-2.0.0-alpha.66/api/docusaurus.config.js.md:118]:
```suggestion
```

not needed

**Comment by slorber** [comment]
thanks, that looks great :) 

No need to doc for alpha 66, as this will only be released in 67.

Also we squash all PRs so your git commit history does not matter much

**Comment by AmyrAhmady** [comment]
Nice! thanks, also I know this isn't the right place to ask, but is there any planned release date for alpha 67

**Comment by slorber** [comment]
we plan to release during the next week

**Comment by zepatrik** [comment]
Released with https://github.com/facebook/docusaurus/releases/tag/v2.0.0-alpha.68

**Pull request merged**